salesforcecli/plugin-source
# fix: add purgeondelete flag to deploy

### What does this PR do?
adds `purgeondelete` to `*:deploy` which will empty the recycle bin when an object is deleted

I'm not sure how to verify this in Lightning

1. create custom object
2. retrieve into local project
3. generate destructive manifest for the custom object (`destructiveChangesPost.xml`)
4. generate a manifest to be deployed alongside the destructive changes, can be empty, or populated (`package.xml`)
5. in the UI, click the astro icon in the top right, and switch to salesforce classic
6. navigate back to setup -> create -> objects
7. run `sfdx force:source:deploy -x package.xml --postdestructivechanges destructiveChangesPost.xml`
8. refresh the page and notice `Deleted Objects(1)` at the bottom of the list
9. click that, and undelete your custom object
10. notice your custom object is back in the list
11. run `sfdx force:source:deploy -x package.xml --postdestructivechanges destructiveChangesPost.xml --purgeondelete`
12. refresh the page and notice the object is deleted, and not available in `Deleted Objects`

### What issues does this PR fix or reference?
@W-4649658@
